rdt-open-reader/src/test/java/io/ona/rdt/robolectric/util/UtilsTest.java

```diff
@@ -223,4 +232,64 @@ public void testIsValidJSONObjectShouldReturnCorrectStatus() {
         Assert.assertFalse(Utils.isValidJSONObject(null));
         Assert.assertFalse(Utils.isValidJSONObject(""));
     }
+
+    @Test
+    public void testVerifyUserAuthorizationShouldExecuteAuthorizationTask() {
+
+        Utils.UserAuthorizationVerificationTask userAuthorizationVerificationTask = Mockito.mock(Utils.UserAuthorizationVerificationTask.class);
+        ReflectionHelpers.setStaticField(Utils.UserAuthorizationVerificationTask.class, "INSTANCE", userAuthorizationVerificationTask);
+
+        Mockito.when(userAuthorizationVerificationTask.getStatus()).thenReturn(AsyncTask.Status.PENDING);
+        Utils.verifyUserAuthorization(RDTApplication.getInstance());
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(1)).execute();
+
+        Mockito.when(userAuthorizationVerificationTask.getStatus()).thenReturn(AsyncTask.Status.RUNNING);
+        Utils.verifyUserAuthorization(RDTApplication.getInstance());
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(1)).execute();
+
+        Mockito.when(userAuthorizationVerificationTask.getStatus()).thenReturn(AsyncTask.Status.FINISHED);
+        Utils.verifyUserAuthorization(RDTApplication.getInstance());
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(1)).destroyInstance();
+        Mockito.verify(userAuthorizationVerificationTask, Mockito.times(2)).execute();
+
+        ReflectionHelpers.setStaticField(Utils.UserAuthorizationVerificationTask.class, "INSTANCE", null);
+    }
+
+    @Test
+    public void testVerifyUserAuthorizationExecutions() throws AuthenticatorException, OperationCanceledException, IOException {
+
+        TestTree testTree = new TestTree();
+        Timber.plant(testTree);
+
+        SyncUtils syncUtils = Mockito.mock(SyncUtils.class);
+        Mockito.when(syncUtils.verifyAuthorization()).thenReturn(false);
+
+        Utils.UserAuthorizationVerificationTask userAuthorizationVerificationTask = Utils.UserAuthorizationVerificationTask.getInstance(RDTApplication.getInstance());
+        ReflectionHelpers.setField(userAuthorizationVerificationTask, "syncUtils", syncUtils);
+
+        userAuthorizationVerificationTask.execute();
+        Mockito.verify(syncUtils, Mockito.times(1)).logoutUser();
+
+        userAuthorizationVerificationTask.destroyInstance();
+        Assert.assertNull(ReflectionHelpers.getStaticField(Utils.UserAuthorizationVerificationTask.class, "INSTANCE"));
+
+        Utils.UserAuthorizationVerificationTask userAuthorizationVerificationTask1 = Utils.UserAuthorizationVerificationTask.getInstance(RDTApplication.getInstance());
+        ReflectionHelpers.setField(userAuthorizationVerificationTask1, "syncUtils", syncUtils);
+
+        Mockito.doThrow(RuntimeException.class).when(syncUtils).logoutUser();
+        userAuthorizationVerificationTask1.execute();
+        Assert.assertEquals(RuntimeException.class.getSimpleName(), testTree.throwable.getClass().getSimpleName());
```

Comment:
This code is repeated above. We only need to mock the syncUtils so that it [throws an exception](https://stackoverflow.com/a/11565324/9782187).

Reply:
fixed.